src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4encoder/HaivisionX4EncoderCommunicator.java

```diff
@@ -1531,26 +2528,439 @@ private void addVideoDataStatisticsToStatisticsProperty(Map<String, String> stat
+		Map<Integer, String> srtModeMap = SRTModeDropdown.getNameToValueMap();
+		Map<Integer, String> stateMap = OutputStateDropdown.getNameToValueMap();
+		Map<Integer, String> encryptionMap = EncryptionDropdown.getNameToValueMap();
+		String protocolOption = null;
+		if (!HaivisionConstant.NONE.equals(srtProtocol)) {
+			protocolOption = protocolMap.get(Integer.parseInt(srtProtocol));
+		}
+		if (protocolOption == null) {
+			protocolOption = srtProtocol;
+		}
+		String bandwidthOverHead =
+				HaivisionConstant.STREAM + HaivisionConstant.SPACE + convertStreamNameUnescapeHtml3(outputResponse.getName()) + HaivisionConstant.HASH + CreateOutputStreamMetric.AVERAGE_BANDWIDTH.getName();
+		for (CreateOutputStreamMetric streamMetric : CreateOutputStreamMetric.values()) {
+			String streamName = HaivisionConstant.STREAM + HaivisionConstant.SPACE + convertStreamNameUnescapeHtml3(outputResponse.getName()) + HaivisionConstant.HASH + streamMetric.getName();
+			switch (streamMetric) {
+				case STATE:
+					String stateStream = outputResponse.getState();
+					value = getNameByValue(stateStream, stateMap);
+					stats.put(streamName, value);
+					break;
+				case CONTENT_NAME:
+					value = checkNoneStringValue(outputResponse.getName());
+					AdvancedControllableProperty contentNameControl = controlText(stats, streamName, convertStreamNameUnescapeHtml3(value));
+					addAdvanceControlProperties(advancedControllableProperties, contentNameControl);
+					break;
+				case DESTINATION_ADDRESS:
+					protocolMode = getNameByValue(protocol, protocolMap);
+					if (!ProtocolDropdown.TS_OVER_SRT.getName().equals(protocolMode)) {
+						value = checkNoneStringValue(outputResponse.getAddress());
+						AdvancedControllableProperty destinationAddress = controlText(stats, streamName, value);
+						addAdvanceControlProperties(advancedControllableProperties, destinationAddress);
+					}
+					break;
+				case DESTINATION_PORT:
+					protocolMode = getNameByValue(protocol, protocolMap);
+					if (!ProtocolDropdown.TS_OVER_SRT.getName().equals(protocolMode)) {
+						value = checkNoneStringValue(outputResponse.getPort());
+						AdvancedControllableProperty destinationPort = controlText(stats, streamName, value);
+						addAdvanceControlProperties(advancedControllableProperties, destinationPort);
+					}
+					break;
+				case PARAMETER_TIMING_AND_SHAPING:
+					value = outputResponse.getShaping().toUpperCase();
+					AdvancedControllableProperty shapingControl = controlDropdown(stats, shapingDropdown, streamName, value);
+					addAdvanceControlProperties(advancedControllableProperties, shapingControl);
+					break;
+				case PARAMETER_BANDWIDTH_OVERHEAD:
+					protocolMode = getNameByValue(protocol, protocolMap);
+					value = outputResponse.getShaping().toUpperCase();
+					String overhead = outputResponse.getBandwidthOverhead();
+					if (ProtocolDropdown.TS_OVER_SRT.getName().equals(protocolMode)) {
+
```

Comment:
please check general code formatting, I see quite a lot of unnecessary whitespaces here or there, maybe just simply autoformatting the file will do the trick :)